pkg/vulnsrc/vulnerability/vulnerability.go

```diff
@@ -56,10 +61,14 @@ func FillAndFilter(vulns []DetectedVulnerability, severities []Severity, ignoreU
 	return vulnerabilities
 }
 
-func getIgnoredIDs() []string {
+func getIgnoredIDs(trivyIgnore string) []string {
 	f, err := os.Open(trivyIgnore)
 	if err != nil {
 		// trivy must work even if no .trivyignore exist
+		if trivyIgnore != DefaultIgnoreFile {
```

Comment:
@knqyf263

I once, remove error handling using if
And, I tested running trivy

so, loop occurred in the following case

1. not exist .trivyignore ( rm .trivyignore )
2. specified not exist ignorefile ( --ignorefile fackignore )

3. os.Open("fackignore") return err
4. err was not nil, so return getIgnoredIDs(DefaultIgnoreFile) 
5. but, .trivyignore not exist 
6. so, os.Open(".trivyignore") return err again
  --> loop occurred


considering the this case, 
I re-add check if trivyIgnore variable is ".trivyignore" or Not
and 
if os.Open(".trivyignore") return err and trivyIgnore variable is not ".trivyignore", this function return nil

Reply:
I meant that we didn't have to check the default path. A user specified a custom path, so the default path should be ignored, I think. In short, the line from 68 to 71 is unnecessary.

Reply:
@knqyf263 
I overlooked the case where the user used this option.
just as you say, we should not adapt .trivyignore when the specified file does not exist
I remove the adaptation of .trivyignore